From vapier@gentoo.org  Tue Dec 22 12:01:32 2009
From: Mike Frysinger <vapier@gentoo.org>
Date: Fri, 18 Dec 2009 16:33:03 -0500
Subject: USB: cypress_m8: allow unstable baud rates
To: Lonnie Mendez <dignome@gmail.com>
Cc: linux-usb@vger.kernel.org
Message-ID: <1261171983-20131-3-git-send-email-vapier@gentoo.org>


I've got a crappy cypress converter here, and while running at higher
baud rates craps out on throughput, it works fine with lower ones.
While it'd be nice to simply use a lower baud rate, not all devices
can be configured this way, and it is possible to (slowly) interact
at higher rates by sending a byte at a time.  So let people force
higher rates when they need it via a module parameter.

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/cypress_m8.c |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/drivers/usb/serial/cypress_m8.c
+++ b/drivers/usb/serial/cypress_m8.c
@@ -73,6 +73,7 @@
 static int debug;
 static int stats;
 static int interval;
+static int unstable_bauds;
 
 /*
  * Version Information
@@ -291,6 +292,9 @@ static int analyze_baud_rate(struct usb_
 	struct cypress_private *priv;
 	priv = usb_get_serial_port_data(port);
 
+	if (unstable_bauds)
+		return new_rate;
+
 	/*
 	 * The general purpose firmware for the Cypress M8 allows for
 	 * a maximum speed of 57600bps (I have no idea whether DeLorme
@@ -1643,3 +1647,5 @@ module_param(stats, bool, S_IRUGO | S_IW
 MODULE_PARM_DESC(stats, "Enable statistics or not");
 module_param(interval, int, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(interval, "Overrides interrupt interval");
+module_param(unstable_bauds, bool, S_IRUGO | S_IWUSR);
+MODULE_PARM_DESC(unstable_bauds, "Allow unstable baud rates");
